NickvisionTubeConverter.Shared/Controllers/MainWindowController.cs

```diff
@@ -172,6 +176,22 @@ public async Task StartupAsync()
                 NotificationSent?.Invoke(this, new NotificationSentEventArgs(_("Unable to unlock keyring. Restart the app to try again."), NotificationSeverity.Error));
             }
         }
+        //Check Network
+        if (!await CheckNetworkConnectivityAsync())
+        {
+            NotificationSent?.Invoke(this, new NotificationSentEventArgs(_("No active internet connection"), NotificationSeverity.Error, "no-network"));
+        }
+    }
+
+    /// <summary>
+    /// Checks for an active network connection
+    /// </summary>
+    /// <returns>True if network connection active, else false</returns>
+    public async Task<bool> CheckNetworkConnectivityAsync()
+    {
+        using var ping = new Ping();
+        var reply = await ping.SendPingAsync("8.8.8.8"); //google
```

Comment:
I wonder how well this is going to work where Google services are fully or partially blocked.
There's [NetworkAvailabilityChanged](https://learn.microsoft.com/en-us/dotnet/api/system.net.networkinformation.networkchange.networkavailabilitychanged?view=net-7.0#system-net-networkinformation-networkchange-networkavailabilitychanged) event that we could probably use and show banner not only on start.

Reply:
The problem is i believe NetworkAvaliablity APIs only tell if your connected to a network, not an active internet connect. So you can be connected to your router's network (LAN) but have no internet connection to the outside world. So it'll say you have a network connection, but you can't access the internet.

So we need a reliable site (could be literally anything) on the internet that we could ping to check if we have active internet. We could even use nickvision.org lol . I just wanted to use something more robust because if said site goes down, it'll say you have no internet.

Reply:
Then I have a different idea: we could ask NetworkManager using Tmds.Dbus. I can try to implement this tonight.

Reply:
The thing is I would like to keep it cross platform for android app and eventually windows...maybe we can just ping gnome.org would work?

Reply:
[Gio.NetworkMonitor](https://docs.gtk.org/gio/method.NetworkMonitor.get_connectivity.html)

Reply:
We're going to go with this solution: https://stackoverflow.com/questions/2031824/what-is-the-best-way-to-check-for-internet-connectivity-using-net/2031831#2031831

To keep windows/android support in backend

Reply:
Well you can still use the NetworkMonitor `network-changed` signal on the GNOME implementation to automatically recheck

Reply:
> Well you can still use the NetworkMonitor `network-changed` signal on the GNOME implementation to automatically recheck post-startup

That would work :)

Reply:
I'm not sure splitting network monitoring in 2 assemblies is a good idea. We could do that instead: if there's at least one active download in downloading state and current total speed is 0, check network using pings every N seconds (5? 10?).

Reply:
> I'm not sure splitting network monitoring in 2 assemblies is a good idea. We could do that instead: if there's at least one active download in downloading state and current total speed is 0, check network using pings every N seconds (5? 10?).

That could work too...but i just remembered csharp's networkinformation itself also provides a network changed event that we can subscribe too and ping to see if it's an active internet connection.